### DISCUSSION TITLE
You say - We check the origin so CSRF check is un-necessary

### DISCUSSION THREAD

**Discussion by imcm7** [Hebilicious/authjs-nuxt#97] [Q&A]
https://github.com/Hebilicious/authjs-nuxt/blob/2100f25fd7de00095e00b330db3c3e29c22d1209/packages/authjs-nuxt/src/runtime/lib/client.ts#L81

When, I can add that path to excludedUrls option of nuxt csrf module?

**Comment by imcm7** [comment]
May be to check if module nuxt-csurf or nuxt-security is installed and add that token, or remove that option completely and not to pass null.

**Comment by joebailey26** [comment]
It would be good to have some discussion around this or documentation on why the CSRF check is not necessary. I'm pretty sure you could spoof an origin header.

**Comment by woldtwerk** [comment]
how exactly do we get the host?
My project works fine locally, but in a docker container on preview server, the /api/auth/callback and redirect url point to localhost:dockerport instead of the baseurl of the app.
I tried adding:

nuxt.config.ts:
authJs: { baseUrl: 'my url' }

.env
NUXT_NEXTAUTH_URL=myurl
NEXTAUTH_URL=myurl

dockerfile:
ENV HOSTNAME 0.0.0.0

nothing is working

**Comment by beeGiaLe** [comment]
I've got the same issue. Tried NUXT_NEXTAUTH_URL, BASE_URL, NUXT_BASE_URL, NEXTAUTH_URL, and it always throws 500: CSRF protected

**Comment by Panca6** [comment]
oh god anyone figure this out yet?
